src/high-level/matrix.lisp

```diff
@@ -443,115 +456,67 @@ If :SQUARE is T, then the result will be restricted to the lower leftmost square
 
-(defgeneric svd (matrix &key reduced)
-  (:documentation "Find the SVD of a matrix M. Return (VALUES U SIGMA Vt) where M = U*SIGMA*Vt")
-  (:method ((matrix matrix) &key reduced)
-    (declare (ignore matrix reduced))
-    (error "SVD is not defined for the generic matrix type.")))
+(define-backend-function inv (matrix)
+  "Get the inverse of the matrix")
 
-(defgeneric qr (matrix)
-  (:documentation "Finds the QR factorization of MATRIX. Returns two matrices (Q, R).
+(define-backend-function svd (matrix &key reduced)
+  "Find the SVD of a matrix M. Return (VALUES U SIGMA Vt) where M = U*SIGMA*Vt")
 
-NOTE: If MATRIX is not square, this will compute the reduced QR factoriation.")
-  (:method ((matrix matrix))
-    (declare (ignore matrix))
-    (error "QR is not defined for the generic matrix type.")))
+(define-backend-function qr (matrix)
+  "Finds the QR factorization of MATRIX. Returns two matrices (Q, R).
 
-(defgeneric ql (matrix)
-  (:documentation "Finds the QL factorization of MATRIX. Returns two matrices (Q, L).
+NOTE: If MATRIX is not square, this will compute the reduced QR factorization.")
 
-NOTE: If MATRIX is not square, this will compute the reduced QL factoriation.")
-  (:method ((matrix matrix))
-    (declare (ignore matrix))
-    (error "QL is not defined for the generic matrix type.")))
+(define-backend-function ql (matrix)
+  "Finds the QL factorization of MATRIX. Returns two matrices (Q, L).
 
-(defgeneric rq (matrix)
-  (:documentation "Finds the RQ factorization of MATRIX. Returns two matrices (R, Q).
+NOTE: If MATRIX is not square, this will compute the reduced QL factorization.")
 
-NOTE: If MATRIX is not square, this will compute the reduced RQ factoriation.")
-  (:method ((matrix matrix))
-    (declare (ignore matrix))
-    (error "RQ is not defined for the generic matrix type.")))
+(define-backend-function rq (matrix)
+  "Finds the RQ factorization of MATRIX. Returns two matrices (R, Q).
 
-(defgeneric lq (matrix)
-  (:documentation "Finds the LQ factorization of MATRIX. Returns two matrices (L, Q).
+NOTE: If MATRIX is not square, this will compute the reduced RQ factorization.")
 
-NOTE: If MATRIX is not square, this will compute the reduced LQ factoriation.")
-  (:method ((matrix matrix))
-    (declare (ignore matrix))
-    (error "LQ is not defined for the generic matrix type.")))
+(define-backend-function lq (matrix)
+  "Finds the LQ factorization of MATRIX. Returns two matrices (L, Q).
+
+NOTE: If MATRIX is not square, this will compute the reduced LQ factorization.")
+
+(define-backend-function expm (matrix)
+  "Computes the exponential of a square matrix M.")
+
+(define-backend-function logm (matrix)
+  "Finds the matrix logarithm of a given square matrix M assumed to be diagonalizable, with nonzero eigenvalues.")
+
+(define-backend-implementation logm :lisp
```

Comment:
Is there also a generic `expm` implementation I am missing?

Reply:
yes, you're right could be implemented by diagonalizing as well; out of scope for this PR tho